resource/httplink/parse.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httplink
+
+import (
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"golang.org/x/xerrors"
+)
+
+const (
+	// https://tools.ietf.org/html/rfc7230#section-3.2.6
+	token  = "[!#$%&'*+\\-.^_`|~0-9A-Za-z]+"
+	quoted = `"(?:[^"\\]|\\.)*"`
+	// https://tools.ietf.org/html/rfc8288#section-3
+	linkParam = token + `\s*(?:=\s*(?:` + token + `|` + quoted + `))?`
+	linkValue = `<([^<>]*)>(?:\s*;\s*` + linkParam + `)*`
```

Comment:
I'm not familiar with regex magics at all, but does it make sense to give each subexp a name (only for readability)? e.g.
```
linkValue = `<(?P<rawURL>[^<>]*)>(?:\s*;\s*` + linkParam + `)*`
...
reValue = regexp.MustCompile(`^\s*(?P<rawLink>` + linkValue + `)\s*(?:,|$)`)
```

Reply:
I don't like adding real constructs to be never used. I added comments to clarify what m[1] and m[2] capture instead.

Reply:
Thanks!